octavia-cli/unit_tests/test_api_http_headers.py

```diff
@@ -3,11 +3,26 @@
 #
 
 import os
+from unittest.mock import MagicMock
 
 import pytest
 from octavia_cli import api_http_headers
 
 
+# Monkeypatch for test
+def new_post_init(self):
+    try:
+        assert isinstance(self.name, str) and self.name
+        assert (isinstance(self.value, str) or isinstance(self.value, MagicMock)) and self.value
+    except AssertionError:
+        raise AttributeError("Header name and value must be non empty string.")
+    self.name = self.name.strip()
+    self.value = self.value.strip()
+
+
+api_http_headers.ApiHttpHeader.__post_init__ = new_post_init
```

Comment:
Sorry to come again with another change request 😄 
There are builtin monkey patching utils in our pytest project (the `mocker` fixture) that you could call directly from the test function:

```python
def test_init(mocker, header_name, header_value, expected_error, expected_name, expected_value)
  mocker.patch.object(api_http_headers.ApiHttpHeader, "__post_init___")
```

this will replace `__post_init__` with a Mock object.

I don't really get why this patching is required though.

Reply:
With the key error beingthat value is a MagicMock:

self = ApiHttpHeader(name='Authorization', value=<MagicMock name='airbyte_api_client.Configuration().get_basic_auth_token()' id='4389064464'>)

Reply:
In 348aefe8ef23632a920514c0b43f5ebf411133db I coerced the header key and value into strings - I think this simplifies things and has the same effect (all the tests still pass)